### PR TITLE
OpenStack: Ensure workflows deprovision credentials

### PR DESCRIPTION
**Pull request by mandre**
For all workflows that include the `openstack-conf-clouds` ref in its `pre` section, also include the `openstack-deprovision-credentials` ref in its `post` section to ensure that created application credentials are cleaned up.

### CODE REVIEW

**Comment by pierreprinetti** [comment]
Your solution would work, but it requires us to always remember to add PASSWORD to the config. Have you considered adding [`openstack-deprovision-credentials`](https://github.com/openshift/release/tree/master/ci-operator/step-registry/openstack/deprovision/credentials) to the `post` stanza of [`shiftstack-quotas-workflow`](https://github.com/openshift/release/blob/f03153fa4ad36c0e10050d977e7f0f7619d2163a/ci-operator/step-registry/shiftstack/quotas/shiftstack-quotas-workflow.yaml#L4) instead?

**Comment by mandre** [comment]
> Your solution would work, but it requires us to always remember to add PASSWORD to the config. Have you considered adding [`openstack-deprovision-credentials`](https://github.com/openshift/release/tree/master/ci-operator/step-registry/openstack/deprovision/credentials) to the `post` stanza of [`shiftstack-quotas-workflow`](https://github.com/openshift/release/blob/f03153fa4ad36c0e10050d977e7f0f7619d2163a/ci-operator/step-registry/shiftstack/quotas/shiftstack-quotas-workflow.yaml#L4) instead?

I have not, and I agree it would be better. Thanks for the tip, I'll update my patch.

**Comment by mandre** [comment]
/pj-rehearse pull-ci-openshift-installer-master-openstack-manifests

**Comment by mandre** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-monitor-mecha-central

**Comment by mandre** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-quotas-vh-mecha-central

File: ci-operator/step-registry/shiftstack/cleanup/shiftstack-cleanup-workflow.yaml [hunk 1]
```diff
@@ -5,5 +5,7 @@ workflow:
     - ref: openstack-conf-clouds
     test:
     - ref: shiftstack-cleanup
+    post:
+    - ref: openstack-deprovision-credentials
```
Comment (pierreprinetti) [inline ci-operator/step-registry/shiftstack/cleanup/shiftstack-cleanup-workflow.yaml:9]: I can't remember why password authentication is mandatory for the cleanup job... but it is, and this means that this step is not necessary

**Comment by pierreprinetti** [comment]
Good job tracking all workflows! I have just left a comment about the cleanup step.

**Review by pierreprinetti — APPROVED** [review]
Approving, as the deprovision step should be a noop when no appcreds are available.

**Pull request merged**